govuk_chat_evaluation/rag_answers/data_models.py

```diff
@@ -10,11 +38,102 @@ class GenerateInput(BaseModel):
+            name = str(uuid.uuid4()),
             expected_output=self.ideal_answer,
             actual_output=self.llm_answer,
+            retrieval_context=[ctx.to_flattened_string() for ctx in self.retrieved_context]
         )
+
+
+class MetricName(str, Enum):
+    FAITHFULNESS = "faithfulness"
+    RELEVANCE = "relevance"
+    BIAS = "bias"
+    # others to add
+
+class LLMJudgeModel(str, Enum):
+    GPT_4O_MINI = "gpt-4o-mini"
+    GPT_4O = "gpt-4o"
+    AMAZON_NOVA_MICRO_1 = "eu.amazon.nova-micro-v1:0"
+    AMAZON_NOVA_PRO_1 = "eu.amazon.nova-pro-v1:0"
+    GEMINI_15_PRO = "gemini-1.5-pro-002"
+    GEMINI_15_FLASH = "gemini-1.5-flash-002"
+
+class LLMJudgeModelConfig(BaseModel):
+    model: LLMJudgeModel
+    temperature: float = 0.0
+    
+    def instantiate_llm_judge(self):
+        """Return the LLM judge model instance."""
+        match self.model:
+            case LLMJudgeModel.AMAZON_NOVA_MICRO_1:
+                raise NotImplementedError(f"Judge model {self.model} instantiation not implemented.")
+                # Placeholder for actual class instance - e.g., CustomAmazonNovaJudge(model_name=self.model.value, temperature=self.temperature)
+            case LLMJudgeModel.AMAZON_NOVA_PRO_1:
+                raise NotImplementedError(f"Judge model {self.model} instantiation not implemented.")
+            case LLMJudgeModel.GEMINI_15_PRO:
+                raise NotImplementedError(f"Judge model {self.model} instantiation not implemented.")
+            case LLMJudgeModel.GEMINI_15_FLASH:
+                raise NotImplementedError(f"Judge model {self.model} instantiation not implemented.")
+            case LLMJudgeModel.GPT_4O_MINI | LLMJudgeModel.GPT_4O:
+                return self.model.value  # Just returns the string name as they are in-built models for DeepEval llm judge
+    
+
+class MetricConfig(BaseModel):
+    name: MetricName
+    threshold: float
+    llm_judge: LLMJudgeModelConfig
+
+    @model_validator(mode="before")
+    @classmethod
+    def inject_llm_judge(cls, values: dict[str, Any]) -> dict[str, Any]:
+        # extract model and temperature to build llm_judge
+        if "llm_judge" not in values:
+            values["llm_judge"] = {
+                "model": values.pop("model"),
+                "temperature": values.pop("temperature", 0.0),
+            }
+        return values
+
+    def to_metric_instance(self):
+        model = self.llm_judge.instantiate_llm_judge()
+        match self.name:
+            case MetricName.FAITHFULNESS:
+                return FaithfulnessMetric(threshold=self.threshold, model=model)
+            case MetricName.RELEVANCE:
+                return AnswerRelevancyMetric(threshold=self.threshold, model=model)
+            case MetricName.BIAS:
+                return BiasMetric(threshold=self.threshold, model=model)
+
+
+class EvaluationConfig(BaseModel):
+    metrics: list[MetricConfig]
+    n_runs: int
+
+    def get_metric_instances(self):
```

Comment:
```suggestion
    def metric_instances(self):
```

nitpick that I don't think you should use a get_ prefix as it makes it look like a getter method and should have same meaning without that loaded prefix.